src/main/java/org/jenkinsci/main/modules/cli/auth/ssh/PublicKeySignatureWriter.java

```diff
@@ -40,25 +43,69 @@ public String asString(PublicKey key) {
     }
 
     public String asString(DSAPublicKey key) {
-        TypesWriter tw = new TypesWriter();
+        PemWriter tw = new PemWriter();
         tw.writeString("ssh-dss");
         DSAParams p = key.getParams();
-        tw.writeMPInt(p.getP());
-        tw.writeMPInt(p.getQ());
-        tw.writeMPInt(p.getG());
-        tw.writeMPInt(key.getY());
+        tw.writeBigInt(p.getP());
+        tw.writeBigInt(p.getQ());
+        tw.writeBigInt(p.getG());
+        tw.writeBigInt(key.getY());
         return encode(tw);
     }
 
     public String asString(RSAPublicKey key) {
-        TypesWriter tw = new TypesWriter();
+        PemWriter tw = new PemWriter();
         tw.writeString("ssh-rsa");
-        tw.writeMPInt(key.getPublicExponent());
-        tw.writeMPInt(key.getModulus());
+        tw.writeBigInt(key.getPublicExponent());
+        tw.writeBigInt(key.getModulus());
         return encode(tw);
     }
 
-    private String encode(TypesWriter tw) {
-        return new String(Base64.encode(tw.getBytes()));
+    private String encode(PemWriter tw) {
+        return Base64.encode(tw.getBytes());
+    }
+
+    private static class PemWriter {
+        private final ByteArrayOutputStream byteOutputStream = new ByteArrayOutputStream();
+
+        private byte[] getBytes() {
+            return byteOutputStream.toByteArray();
+        }
+
+        private void writeString(String value) {
+            byte[] bytes = value.getBytes(StandardCharsets.ISO_8859_1);
```

Comment:
I was keeping the encoding used by the Trilead writer that this class replaces. I'm not there's a particular need to do so though, and UTF_8 or ASCII would probably work just as well.